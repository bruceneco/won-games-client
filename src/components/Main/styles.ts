import styled from 'styled-components'

export const Wrapper = styled.main`
  background: #06092b;
  height: 100%;
  width: 100%;
  color: white;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 3rem;
  text-align: center;
`

export const Logo = styled.img`
  width: 25rem;
`

export const Title = styled.h1`
  margin-top: 2rem;
  font-size: 2.5rem;
`

export const Description = styled.h2`
  font-size: 2rem;
  font-weight: 400;
  margin-top: 1.5rem;
`

export const Illustration = styled.img`
  width: min(30rem, 100%);
  margin-top: 3rem;
`
