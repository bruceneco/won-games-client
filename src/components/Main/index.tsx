import * as S from "./styles";

const Main = ({
  title = "NextJS Boilerplate",
  description = "TypeScript, ReactJS, NextJS and Styled-Components"
}): JSX.Element => (
  <S.Wrapper>
    <S.Logo
      src="img/logo.svg"
      alt="Atom illustration representing react framework"
    />
    <S.Title>{title}</S.Title>
    <S.Description>{description}</S.Description>
    <S.Illustration
      src="img/hero-illustration.svg"
      alt="Programmer in front of code blocks"
    />
  </S.Wrapper>
);

export default Main;
