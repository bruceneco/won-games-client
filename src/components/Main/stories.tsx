import Main from ".";
import { Meta, Story } from "@storybook/react";

export default {
  title: "Main",
  component: Main,
  args: {
    title: "Test default",
    description: "Default also"
  }
} as Meta;

export const Basic: Story = (args) => <Main {...args} />;

export const Default: Story = (args) => <Main {...args} />;
Default.args = {
  title: "NextJS Boilerplate",
  description: "TypeScript, NextJS, ReactJS and Styled-Components"
};
