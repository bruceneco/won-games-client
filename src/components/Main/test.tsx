import { render, screen } from "@testing-library/react";
import Main from "./index";

describe("<Main/>", () => {
  it("render heading", () => {
    render(<Main />);
    expect(
      screen.queryByRole("heading", { name: /NextJS Boilerplate/i })
    ).toBeInTheDocument();
  });
  it("snapshot is match", () => {
    const { container } = render(<Main />);
    expect(container.firstChild).toMatchSnapshot();
  });
  it("right background color", () => {
    const { container } = render(<Main />);
    expect(container.firstChild).toHaveStyle({ "background-color": "#06092b" });
  });
});
