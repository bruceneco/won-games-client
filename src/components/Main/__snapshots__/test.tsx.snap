// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main/> snapshot is match 1`] = `
<main
  class="styles__Wrapper-heyFP xXCWS"
>
  <img
    alt="Atom illustration representing react framework"
    class="styles__Logo-bSryDj kvMBXl"
    src="img/logo.svg"
  />
  <h1
    class="styles__Title-cLvJfs eQmsKe"
  >
    NextJS Boilerplate
  </h1>
  <h2
    class="styles__Description-elWOBK bEakSu"
  >
    TypeScript, ReactJS, NextJS and Styled-Components
  </h2>
  <img
    alt="Programmer in front of code blocks"
    class="styles__Illustration-fyJZJM cSFNml"
    src="img/hero-illustration.svg"
  />
</main>
`;
